src/components/events-table.js

```diff
@@ -59,25 +64,145 @@ function inputDataRow(processEvent) {
+        setLogFilter(event.currentTarget.value.toUpperCase());
+        filterProcessEvent(
+            levelFilter,
+            event.currentTarget.value.toUpperCase()
+        );
+    };
+
+    const filterProcessEvent = (currentEventFIlter, currentLogFilter) => {
+        let filtered;
+        filtered = processEvents.filter(
+            (event) =>
+                event.level.includes(currentEventFIlter) &&
+                event.message.toUpperCase().includes(currentLogFilter)
+        );
+        setFiltredProcessEvents(filtered);
+    };
+
+    const handleCloseLevel = () => {
+        setAnchorLevelEl(null);
+    };
+    const handleCloseLog = () => {
+        setAnchorLogEl(null);
+    };
+
     return (
         <TableContainer component={Paper}>
             <Table className="table">
                 <TableHead>
                     <TableRow>
                         <TableCell>
                             <FormattedMessage id="level" />
+                            <span>
+                                <Button
+                                    aria-controls="simple-menu"
+                                    aria-haspopup="true"
+                                    onClick={handleLevelMenuClick}
+                                    style={{
+                                        color:
+                                            logFilter === ''
+                                                ? 'inherit'
+                                                : '#3F51b5',
+                                    }}
+                                >
+                                    <FilterList />
+                                </Button>
+                                <Menu
+                                    id="simple-menu"
+                                    anchorEl={anchorLevelEl}
+                                    keepMounted
+                                    open={Boolean(anchorLevelEl)}
+                                    onClose={handleCloseLevel}
+                                >
+                                    <MenuItem>
+                                        <TextField
+                                            id="levelTextField"
+                                            label={
+                                                <FormattedMessage id="filterOnLevel" />
+                                            }
+                                            type="text"
+                                            defaultValue={levelFilter}
+                                            InputLabelProps={{
+                                                shrink: true,
+                                            }}
+                                            onChange={handleLevelChange}
+                                        />
+                                    </MenuItem>
+                                </Menu>
+                            </span>
```

Comment:
I think you can extract a component for that to make it common for level and message

Reply:
I extracted the filter menu and create a new component. I also use the new component in global-view-core